### PR TITLE
Clean up some typing syntax

### PR DESCRIPTION
**Pull request by Sachaa-Thanasius**
Ideally, this doesn't result in any change to the functionality; it's more about making the typing syntax consistent with python 3.10+ and ensuring pyright flags it accordingly if it isn't.

### CODE REVIEW

File: pyproject.toml
Comment (AbstractUmbra) [inline pyproject.toml:52]: Since the ruleset is strict, this is unnecessary.

File: wavelink/node.py
Comment (AbstractUmbra) [inline wavelink/node.py:69]: Triple check this is valid in use. There has been some issues where the union type needs to be stringified to work correctly. 
E.g. `"A | B | C"`

**Review by AbstractUmbra — CHANGES_REQUESTED** [review]
I know this is still in draft but some things to note so far.

File: pyproject.toml
Comment (Sachaa-Thanasius) [reply pyproject.toml:52]: These two rules are off even in strict, to be fair, but that's valid. I can remove them.

File: wavelink/node.py
Comment (Sachaa-Thanasius) [reply wavelink/node.py:69]: This should work since future annotations is imported, I think, but I can put `TypeAlias` in front of it just in case and try to run a simple bot, see if it breaks that way.

**Comment by Sachaa-Thanasius** [comment]
Maybe more could be done, but this feels like it could be enough for a PR that doesn't add functionality.

**Review by AbstractUmbra — APPROVED** [review]
Looks good to me, waiting for @EvieePy for final review.

**Comment by EvieePy** [comment]
Has this been tested in terms of actually being able to be ran?

**Comment by Sachaa-Thanasius** [comment]
> Has this been tested in terms of actually being able to be ran? With like simple.py or something?

Yep. It works fine with simple.py on my machine, though I didn't run it exhaustively — I loaded up "Ocean Drive" and a "Rock Classics" Spotify playlist (courtesy of the examples in the migrating docs) and just let it go for a while.

File: wavelink/player.py
Comment (EvieePy) [inline wavelink/player.py:49]: Why are we not importing the other classes?

Comment (EvieePy) [reply wavelink/player.py:49]: Nevermind it's early and I saw this was player py not init

**Review by EvieePy — APPROVED** [review]

**Pull request merged**
